kinto/views/admin.py

```diff
@@ -0,0 +1,134 @@
+"""
+Special views for administration.
+"""
+
+import itertools
+import colander
+
+from kinto.authorization import RouteFactory
+from kinto.core import utils as core_utils, resource
+from kinto.core.events import notify_resource_event, ACTIONS
+from kinto.core.resource import viewset
+from kinto.core.storage import Filter, Sort
+from kinto.core.storage.utils import paginated
+
+
+def slice_into_batches(iterable, batch_size):
+    # Taken from https://code.activestate.com/recipes/303279-getting-items-in-batches/
+    i = iter(iterable)
+    while True:
+        batchiter = itertools.islice(i, batch_size)
+        yield itertools.chain([next(batchiter)], batchiter)
+
+
+class Deleted(resource.ResourceSchema):
+    principal = colander.SchemaNode(colander.String())
+
+    class Options:
+        preserve_unknown = False
+
+
+class UserDataFactory(RouteFactory):
+    def _find_required_permission(self, request, service):
+        return ('user_data', 'delete')
+
+
+class UserDataViewSet(viewset.ShareableViewSet):
+    factory = UserDataFactory
+
+
+@resource.register(name='user-data',
+                   description='Delete the data owned by a user',
+                   collection_path='/__user_data__',
+                   record_path='/__user_data__/{{principal}}',
+                   viewset=UserDataViewSet(),
+                   collection_methods=('DELETE',))
+class UserData(resource.ShareableResource):
+
+    schema = Deleted
+
+    def delete(self):
+        principal = self.request.matchdict['principal']
+        storage = self.request.registry.storage
+        permission = self.request.registry.permission
+        object_uris_and_permissions = permission.get_accessible_objects([principal])
+        object_uris = list(object_uris_and_permissions.keys())
+        object_uris.sort()
```

Comment:
This `sort()` turns out not to be necessary, I misread how the `get_objects_permissions` function works.